### PR TITLE
(GH-36) Restructure documentation

### PR DESCRIPTION
**Pull request by gep13**
Based heavily on what Cake.Issues.Recipe has done.  Thanks @pascalberger

Relates to #36

### CODE REVIEW

File: docs/input/docs/contributing/how-to-build.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+```powershell
+./build.ps1
+```
+
+On OSX/Linux run:
+
+```bash
+./build.sh
+```
```
Comment (AdmiringWorm) [inline docs/input/docs/contributing/how-to-build.md:18]: I would recommend having a note mentioning not to run the `setup.cake` script directly, as we do some additional changes in the bootstrappers to make the dogfooding work as expected.

File: docs/input/docs/index.cshtml [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
 Title: Documentation
+Order: 10
 ---
-<p>This user guide, like Cake.Recipe itself, is under very active development. Some parts of it aren't 
-documented as completely as they need to be, but we gladly accept your contributions.</p>
-
-<p>We need your help to improve the documentation for Cake.Recipe, so if there is something that you 
-would like to add then you can edit the content directly on GitHub.</p>
+<p>
+    We gladly accept your contributions. If there is something that you would like to add then you can edit the content directly on GitHub.
```
Comment (AdmiringWorm) [inline docs/input/docs/index.cshtml:6]: Only partially true, they can edit on GitHub but they would need to open a PR to have any updates being available on the documentation site.
Maybe linking to `how to contribute` documentation as well?

File: docs/input/docs/contributing/how-to-contribute.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+To contribute:
+
+* Fork this repository.
+* Create a feature branch from `develop`.
+* Implement your changes.
+* Push your feature branch.
+* Create a pull request.
+
+For getting started see issues marked with `Up-for-grabs` in the individual repositories.
```
Comment (AdmiringWorm) [inline docs/input/docs/contributing/how-to-contribute.md:17]: We only have 1 repository for Cake.Recipe, and we don't really use the `Up for grabs` label.
I haven't seen it used at least.

**Comment by gep13** [comment]
@AdmiringWorm I believe I have addressed your comments here as well.   Would be good to get this merged in, so that we can continue making small PR's to update docs, rather than one massive PR getting out of hand.

**Pull request merged**